templates/documents/shared/details_base.html

```diff
@@ -0,0 +1,28 @@
+<form
+    id="detail-form"
+    action="{{ edit_url }}"
+    method="POST"
+    class="m-2 grid grid-cols-4" x-data="{editing: false}">
+    {% csrf_token %}
+    {% block detail_content %}
+    {% endblock %}
+    <div class="py-3 col-span-1">
+        <div class="join">
+            <button type="button" class="btn btn-sm join-item pl-3" x-show="!editing" @click="editing = true">
+                Edit <i class="fa-solid fa-pencil fa-xs"></i>
+            </button>
+            <button type="submit" class="btn btn-sm join-item" x-cloak x-show="editing">
+                Update <i class="fa-solid fa-check fa-xs"></i>
+            </button>
+            <a type="button" class="btn btn-sm join-item" x-show="!editing" href="{{ delete_url }}">
+                <i class="fa-solid fa-trash htmx-hide"></i>
```

Comment:
If you're going to use `htmx-hide` then there should also be an element with `htmx-show`:
```suggestion
                <i class="fa-solid fa-trash fa-xs htmx-hide"></i>
                <span class="loading loading-spinner loading-xs htmx-show"></span>
```

Reply:
Addressed in https://github.com/dimagi/open-chat-studio/pull/1317